Framework/NexusGeometry/test/NexusGeometrySaveTest.h

```diff
@@ -284,16 +287,147 @@ class NexusGeometrySaveTest : public CxxTest::TestSuite {
+
+    // translate source
+    Eigen::Vector3d absoluteBankLocation(5, 0, 0);
+
+    compInfo.setPosition(compInfo.indexOfAny("bank1"),
+                         Mantid::Kernel::toV3D(absoluteBankLocation));
+
+    auto relativeSourcePosition = Mantid::Kernel::toVector3d(
+        compInfo.relativePosition(compInfo.indexOfAny("bank1")));
+
+    TS_ASSERT(relativeSourcePosition.isApprox(absoluteBankLocation));
+  }
+
+
+
+  void test_rotation() {
+
+    auto instrument =
+        ComponentCreationHelper::createTestInstrumentRectangular2(1, 10);
+
+    auto beamline =
+        Mantid::Geometry::InstrumentVisitor::makeWrappers(*instrument);
+    auto &compInfo = (*beamline.first);
+
+    Eigen::Quaterniond absoluteBankRotation(
+        Eigen::AngleAxisd(M_PI / 4, Eigen::Vector3d(0, 1, 0)));
+
+    compInfo.setRotation(
+        compInfo.indexOfAny("bank1"), // bank1 defined as in
+                                      // ComponentCreationHelper.cpp
+                                      // line 584 01.07.19
+        Mantid::Kernel::toQuat(absoluteBankRotation));
+
+    auto relativeBankRotation = Mantid::Kernel::toQuaterniond(
+        compInfo.relativeRotation(compInfo.indexOfAny("bank1")));
+
+    TS_ASSERT(relativeBankRotation.isApprox(absoluteBankRotation));
+  }
+
+  void
+  test_bank_absolute_rotation_is_equal_to_detector_in_bank_absolute_rotation() {
+
+    auto instrument =
+        ComponentCreationHelper::createTestInstrumentRectangular2(1, 10);
+
+    auto beamline =
+        Mantid::Geometry::InstrumentVisitor::makeWrappers(*instrument);
+    auto &compInfo = (*beamline.first);
+
+    // set abosolute bank rotation (rotate about y)
+    Eigen::Quaterniond absoluteBankRotation(
+        Eigen::AngleAxisd(M_PI / 4, Eigen::Vector3d(0, 1, 0)));
+
+    Eigen::Quaterniond zeroRotation(
+        Eigen::AngleAxisd(0, Eigen::Vector3d(0, 0, 0)));
+
+    // relative position of detector
+    auto relativeDetectorRotation =
+        Mantid::Kernel::toQuaterniond(compInfo.relativeRotation(0));
+
+    TS_ASSERT(relativeDetectorRotation.isApprox(zeroRotation));
+  }
+
+
+  void test_detector_position_is_expected_value_after_rotation() {
+
+    // 45 degree rotation around z.
+    const Quat bankRotation(45, V3D(0, 0, 1));
+    const Quat detRotation(45, V3D(0, 0, 1));
+
+    auto instrument =
+        ComponentCreationHelper::createSimpleInstrumentWithRotation(
+            Mantid::Kernel::V3D(0, 0, -10), Mantid::Kernel::V3D(0, 0, 0),
+            Mantid::Kernel::V3D(0, 0, 10), bankRotation, detRotation);
+
+    auto beamline =
+        Mantid::Geometry::InstrumentVisitor::makeWrappers(*instrument);
+    const auto &compInfo = *std::get<0>(beamline);
+
+    const auto actualBankRotation =
+        compInfo.rotation(compInfo.indexOfAny("detector-stage"));
+    const auto actualDetRotation = compInfo.rotation(0);
+
+    TS_ASSERT_EQUALS(actualBankRotation, bankRotation);
+    TS_ASSERT_EQUALS(actualDetRotation, detRotation * bankRotation);
+  }
```

Comment:
In this test, you are currently not hooked into any method or functionality from `NexusSaveGeometry`. What are you intending to demonstrate here?